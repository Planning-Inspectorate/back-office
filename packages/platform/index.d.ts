import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
