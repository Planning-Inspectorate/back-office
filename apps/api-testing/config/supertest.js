// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
