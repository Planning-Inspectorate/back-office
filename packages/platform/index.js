export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
