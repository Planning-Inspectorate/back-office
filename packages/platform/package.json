{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
