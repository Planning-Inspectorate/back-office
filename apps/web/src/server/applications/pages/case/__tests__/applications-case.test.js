import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestApplication } from '../../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestApplication();
const request = supertest(app);
const successGetResponse = { id: 1, applicants: [{ id: 1 }] };

const nocks = () => {
	nock('http://test/').get('/applications/case-officer').reply(200, []);
	nock('http://test/')
		.get(/\/applications\/123(.*)/g)
		.times(2)
		.reply(200, successGetResponse);
};

describe('applications view case summary', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	beforeEach(async () => {
		await request.get('/applications-service/case-officer');
	});

	const baseUrl = '/applications-service/create-new-case/123';

	describe('GET /case', () => {
		beforeEach(async () => {
			nocks();
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});

	describe('GET /create-new-case/project-information', () => {
		beforeEach(async () => {
			nocks();
		});

		it('should render the page', async () => {
			const response = await request.get(`${baseUrl}/project-information`);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});
});
