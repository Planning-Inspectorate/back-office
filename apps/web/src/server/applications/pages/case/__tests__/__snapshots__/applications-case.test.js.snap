// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications view case summary GET /case/123 should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row govuk-!-padding-top-9 govuk-!-padding-bottom-2 govuk-!-margin-bottom-7 pins-masthead-summary\\">
        <div class=\\"govuk-grid-column-full\\">
            <h4 class=\\"govuk-heading-m\\"> CASE/04 </h4>
            <h1 class=\\"govuk-heading-xl\\">Case with no sector</h1>
            <div class=\\"govuk-grid-row\\">
                <div class=\\"govuk-grid-column-full govuk-body\\">,
                    <br>
                    <br><strong class=\\"govuk-tag govuk-tag--\\">draft</strong>
                </div>
            </div>
        </div>
    </div>
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-one-third\\">
            <aside class=\\"govuk-!-margin-bottom-8 pins-list-menu\\">
                <nav>
                    <h3 class=\\"govuk-visually-hidden\\">Case Details Menu</h3>
                    <ul class=\\"gov-list pins-list-menu\\">
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link pins-selected-item\\" href=\\"/applications-service/case/4/\\"> Overview</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/project-information/\\"> Project information</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/key-dates/\\"> Key dates</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/project-team/\\"> Project team</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/fees/\\"> Fees</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/project-documentation/\\"> Project documentation</a>
                        </li>
                    </ul>
                </nav>
            </aside>
        </div>
        <div class=\\"govuk-grid-column-two-thirds\\">
            <h2 class=\\"govuk-heading-m\\">Latest key dates</h2>
            <ul class=\\"govuk-list govuk-list--bullet\\">
                <li>Anticipated submission date:</li>
                <li>Anticipated submission date internal:</li>
            </ul><a class=\\"govuk-link\\" href=\\"#\\">Update key dates</a>
            <hr class=\\"govuk-section-break govuk-section-break--l govuk-section-break--visible\\">
            <h2 class=\\"govuk-heading-m\\">Summary information</h2>
            <dl class=\\"govuk-summary-list govuk-summary-list--no-border\\">
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Case reference</dt>
                    <dd class=\\"govuk-summary-list__value\\">CASE/04</dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Applicant Information</dt>
                    <dd class=\\"govuk-summary-list__value\\">Org name
                        <br>email@email.co
                        <br>001</dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Applicant website</dt>
                    <dd class=\\"govuk-summary-list__value\\"><a class=\\"govuk-link\\" href=\\"//website.web\\">website.web</a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Project email</dt>
                    <dd class=\\"govuk-summary-list__value\\"></dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Project page</dt>
                    <dd class=\\"govuk-summary-list__value\\"><strong class=\\"govuk-tag govuk-tag--grey\\">NOT PUBLISHED</strong>
                    </dd>
                </div>
            </dl><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/project-information/\\"> Update project information</a>
            <hr             class=\\"govuk-section-break govuk-section-break--l govuk-section-break--visible\\">
                <h2 class=\\"govuk-heading-m\\">Project team</h2>
                <p class=\\"govuk-body\\">No project members have been added yet</p>
        </div>
    </div>
</main>"
`;

exports[`applications view case summary GET /case/123/project-information should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row govuk-!-padding-top-9 govuk-!-padding-bottom-2 govuk-!-margin-bottom-7 pins-masthead-summary\\">
        <div class=\\"govuk-grid-column-full\\">
            <h4 class=\\"govuk-heading-m\\"> CASE/04 </h4>
            <h1 class=\\"govuk-heading-xl\\">Case with no sector</h1>
            <div class=\\"govuk-grid-row\\">
                <div class=\\"govuk-grid-column-full govuk-body\\">,
                    <br>
                    <br><strong class=\\"govuk-tag govuk-tag--\\">draft</strong>
                </div>
            </div>
        </div>
    </div>
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-one-third\\">
            <aside class=\\"govuk-!-margin-bottom-8 pins-list-menu\\">
                <nav>
                    <h3 class=\\"govuk-visually-hidden\\">Case Details Menu</h3>
                    <ul class=\\"gov-list pins-list-menu\\">
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/\\"> Overview</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link pins-selected-item\\" href=\\"/applications-service/case/4/project-information/\\"> Project information</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/key-dates/\\"> Key dates</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/project-team/\\"> Project team</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/fees/\\"> Fees</a>
                        </li>
                        <li class=\\"govuk-heading-s\\"><a class=\\"govuk-link \\" href=\\"/applications-service/case/4/project-documentation/\\"> Project documentation</a>
                        </li>
                    </ul>
                </nav>
            </aside>
        </div>
        <div class=\\"govuk-grid-column-two-thirds\\">
            <h2 class=\\"govuk-heading-m\\">Project information</h2>
            <dl class=\\"govuk-summary-list govuk-summary-list--no-border\\">
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Project page status</dt>
                    <dd class=\\"govuk-summary-list__value\\"><strong class=\\"govuk-tag govuk-tag--grey\\">Not Published</strong>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Case reference number</dt>
                    <dd class=\\"govuk-summary-list__value\\">CASE/04</dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Sector</dt>
                    <dd class=\\"govuk-summary-list__value\\"></dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Subsector</dt>
                    <dd class=\\"govuk-summary-list__value\\"></dd>
                </div>
            </dl>
            <div class=\\"govuk-accordion\\" data-module=\\"govuk-accordion\\" id=\\"\\">
                <div class=\\"govuk-accordion__section \\">
                    <div class=\\"govuk-accordion__section-header\\">
                        <h2 class=\\"govuk-accordion__section-heading\\"><span class=\\"govuk-accordion__section-button\\" id=\\"-heading-1\\"> Project details</span></h2>
                    </div>
                    <div id=\\"-content-1\\" class=\\"govuk-accordion__section-content\\" aria-labelledby=\\"-heading-1\\">
                        <table class=\\"govuk-table\\">
                            <tbody class=\\"govuk-table__body\\">
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Project name</th>
                                    <td class=\\"govuk-table__cell\\">Case with no sector</td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/name/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Project description</th>
                                    <td class=\\"govuk-table__cell\\">Case with no sector description</td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/description/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Project email address</th>
                                    <td class=\\"govuk-table__cell\\"></td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/team-email/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Project location</th>
                                    <td class=\\"govuk-table__cell\\"></td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/project-location/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Grid references</th>
                                    <td class=\\"govuk-table__cell\\">
                                        <br>
                                    </td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/grid-references/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Regions(s)</th>
                                    <td class=\\"govuk-table__cell\\"></td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/regions/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Map zoom level</th>
                                    <td class=\\"govuk-table__cell\\"></td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/zoom-level/\\"> Change</a>
                                    </td>
                                </tr>
                            </tbody>
                        </table>
                    </div>
                </div>
                <div class=\\"govuk-accordion__section \\">
                    <div class=\\"govuk-accordion__section-header\\">
                        <h2 class=\\"govuk-accordion__section-heading\\"><span class=\\"govuk-accordion__section-button\\" id=\\"-heading-2\\"> Applicant information</span></h2>
                    </div>
                    <div id=\\"-content-2\\" class=\\"govuk-accordion__section-content\\" aria-labelledby=\\"-heading-2\\">
                        <table class=\\"govuk-table\\">
                            <tbody class=\\"govuk-table__body\\">
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Organisation name</th>
                                    <td class=\\"govuk-table__cell\\">Org name</td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/applicant-organisation-name/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Contact name</th>
                                    <td class=\\"govuk-table__cell\\">Lorem Ipsum</td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/applicant-full-name/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Address</th>
                                    <td class=\\"govuk-table__cell\\">Applicant address, ABC123, London</td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/applicant-address/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Website</th>
                                    <td class=\\"govuk-table__cell\\">website.web</td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/applicant-website/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Email address</th>
                                    <td class=\\"govuk-table__cell\\">email@email.co</td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/applicant-email/\\"> Change</a>
                                    </td>
                                </tr>
                                <tr class=\\"govuk-table__row\\">
                                    <th scope=\\"row\\" class=\\"govuk-table__header\\">Telephone number</th>
                                    <td class=\\"govuk-table__cell\\">001</td>
                                    <td class=\\"govuk-table__cell text-align--right pins-table__cell--s\\"><a class=\\"govuk-link\\" href=\\"/applications-service/case/4/edit/applicant-telephone-number/\\"> Change</a>
                                    </td>
                                </tr>
                            </tbody>
                        </table>
                    </div>
                </div>
            </div>
        </div>
    </div>
</main>"
`;
